### DISCUSSION TITLE
Proper way of pause and resume a track

### DISCUSSION THREAD

**Discussion by vvalentinich** [paullouisageneau/libdatachannel#1163] [Q&A] [answered]
I am struggling with pause and resume of a track. For pausing I tried something like removing a track completely, and saw a proper SDP. For resuming I've tried to add a new track with the same ssrc, msid, trackId, etc.. But after adding the new track and calling setLocalDescription I see in a log something like:
`rtc::impl::PeerConnection::processLocalDescription::<lambda_2016d464866ec9eaf569ce159197f09e>::operator ()@923: Adding media to local description, mid="video-3145733", removed=true (track is destroyed)`
I see methods like close(), markRemoved(), but see nothing opposite.

**Comment by paullouisageneau** [comment, marked as answer]
Removal is irreversible in WebRTC media negotiation, so you can't re-add the same track later. It is typically used to reject a track.

Track negotiation is designed only to negotiate formats and codecs, you should not attempt to signal stop and restart with it. You should need renegotiation only when a new media is added later, like screen sharing.

To pause a track, just stop sending, and to resume, restart sending. You can send your own signaling messages to inform the remote side explicitly of the playback state.

**Reply by vvalentinich** [reply]
Ok, thanks. I've done as you describe.
